AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncList+Paginatable.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+
+extension AppSyncList {
+    func nextPage(completion: @escaping PageResultCallback) {
+        guard let nextToken = nextToken else {
+            completion(.failure(CoreError.validation("There is no next page to fetch.",
```

Comment:
[on Android](https://github.com/aws-amplify/amplify-android/blob/6699348354feda0c9749b3c0155a51626ce6defd/core/src/main/java/com/amplifyframework/api/graphql/PaginatedResult.java#L73), 

getNextRequest would return null, so that the developer cannot perform the actual request to get next page. Given this, i think we should continue to return immediately as a failure, as the success result would be confusing